### DISCUSSION TITLE
Be able to search for a definition in the library from double-clicking the canvas

### DISCUSSION THREAD

**Discussion by nicolaasburgers** [nicolaasburgers/definition-library-releases#9] [Ideas]
This idea came out of a meeting with Enzyme APD: it would be great to be able to double-click the canvas and search for a definition from the library to add to the canvas as cluster by name in the same way as you would search for a component.

The feasibility of this would need investigation.

**Comment by jakubczajczynski** [comment]
This is a good idea, searching for clusters in the current GUI (whose filters reset after each closing) is not particularly convenient. Maybe in some cases the full GUI is ok, but when we know exactly what we are looking for and under what name, then quickly reaching this cluster is very important.

**Comment by jakubczajczynski** [comment]
Since this time, my collection of definitions and scripting components has grown a bit. It would be very beneficial if we could checkmark scripting components/clusters that should be accessible from the canvas by double-clicking. Including all of them would bloat the search significantly, so it's not desirable. I have several scripting components collections that are not mature yet to keep as grasshopper plugins but I'm testing them in my work and I would like to have quick access to them.
